src/components/DeviceListTable.tsx

```diff
@@ -0,0 +1,98 @@
+import { FC } from "react";
+import { MainTable } from "@canonical/react-components";
+import { isRootDisk } from "util/instanceValidation";
+import { FormDevice } from "util/formDevices";
+import ResourceLink from "components/ResourceLink";
+import { isOtherDevice } from "util/devices";
+import OverviewPageDeviceDetails from "./OverviewPageDeviceDetails";
+import { useParams } from "react-router-dom";
+import { LxdDeviceValue, LxdDevices } from "types/device";
+
+interface Props {
+  configBaseURL: string;
+  devices: LxdDevices;
+}
+
+const DeviceListTable: FC<Props> = ({ configBaseURL, devices }) => {
+  const { project } = useParams<{ project: string }>();
+
+  const compareDevices = (
+    a: [string, LxdDeviceValue],
+    b: [string, LxdDeviceValue],
+  ) => {
+    const nameA = a[0].toLowerCase();
+    const nameB = b[0].toLowerCase();
+    const typeA = a[1].type;
+    const typeB = b[1].type;
+    if (typeA === typeB) {
+      return nameA.localeCompare(nameB);
+    }
+
+    return typeA.localeCompare(typeB);
+  };
+
+  // Identify non-NIC devices
+  const overviewDevices = Object.entries(devices ?? {})
+    .filter(([_key, device]) => {
+      return device.type !== "nic" && device.type !== "none";
+    })
+    // Sort devices by type and name
+    .sort(compareDevices);
+
+  const hasDevices = overviewDevices.length > 0;
+
+  const deviceHeaders = [
+    { content: "Name", sortKey: "name", className: "u-text--muted" },
+    { content: "Type", sortKey: "type", className: "u-text--muted" },
+    { content: "Details", className: "u-text--muted" },
+  ];
+
+  const deviceRows = overviewDevices.map(([devicename, device]) => {
+    return {
+      columns: [
+        {
+          content: (
+            <ResourceLink
+              type="device"
+              value={devicename}
+              to={`${configBaseURL}/${isOtherDevice(device) ? "other" : device.type}`}
+            />
+          ),
+          role: "cell",
+          "aria-label": "Name",
+        },
+        {
+          content: `${device.type} ${isRootDisk(device as FormDevice) ? "(root)" : ""}`,
+
+          role: "cell",
+          "aria-label": "Type",
+        },
+        {
+          content: (
+            <OverviewPageDeviceDetails
+              device={device}
+              project={project as string}
+            />
+          ),
+          role: "cell",
+          "aria-label": "Details",
+        },
+      ],
+      sortData: {
+        name: devicename.toLowerCase(),
+        type: device.type,
+      },
+    };
+  });
+
+  return (
+    <>
+      {hasDevices && (
+        <MainTable headers={deviceHeaders} rows={deviceRows} sortable />
```

Comment:
This table is missing the class "devices", probably from a wrong resolution of the merge conflicts? With the class the space among the column should be distributed in a better way, avoiding spread to two lines and aligning with of the first two columns with the tables above/below.

Reply:
Actually just applying the class is probably wrong, because the styling is applied only for instance overview page then and not for profile overview. 
I think best is to apply a "device-table" class here, and define a new css file. Then move the "devices" css block from the _instance_detail_overview.scss to the new file, to the root level, so it will be applied on both pages for instances and devices.